### DISCUSSION TITLE
Add ability to disable right clicking on iframe. Disable contextmenu.

### DISCUSSION THREAD

**Discussion by lt692** [vimeo/player.js#799] [Ideas]
### Expected Behavior
Developer can disable contextmenu for iframe
### Actual Behavior
Developer can't disable contextmenu and user can simply copy vimeo iframe URL and share with other who can view click easily outside of website.
### Steps to Reproduce
Just add iframe to your html and try rightclicking it to reveal contextmenu.

Tried to disable developer tool on whole page, still iframe does not care that rest of page contextmenu is disabled and still allows user to inspect page source and grab URL. 

Would be nice feature to have that allows to intercept context menu -> player.on("contextmenu", function (e) {
e.preventDefault();
});

**Comment by fisherinnovation** [comment]
Thanks for your feature request! I am moving this thread to the discussion board.

**Reply by floinay** [reply]
?

**Comment by kh1rfan08** [comment]
Bumping this as users are able to access the url to share what's meant to be paid trainings via just a right click in iframe

**Reply by bdougherty** [reply]
@kh1rfan08 if you are looking to restrict access, your best bet here is to set the allowed domains for embedding. Disabling the context menu will not prevent people from being able to get the embed url.

https://help.vimeo.com/hc/en-us/articles/13806764653201-How-to-change-your-video-s-privacy-settings#h_01GQ891KDXKVFFZM7T6C4P4NZ3

**Reply by kh1rfan08** [reply]
@bdougherty won't prevent advanced users no, our customers however are basic, but to correct any simplicity in just right click copy pasting the url to other consumers who haven't paid for the resource it seemed like a viable solution to offer context-menu restrictions from the parent...

I've put in an overlay that uses z-index to omit/capture right-clicks and return false on events from overtop of the iframe as a "solution" until a better hosting option can be found

**Reply by phuryn** [reply]
That's a no-brainer and should be implemented @fisherinnovation 
Exactly the case @kh1rfan08 describes

It won't prevent the advanced users, but getting the URL with the help of Dev Tools is different than presenting this URL in the context menu. People may not even understand they are sharing a paid resource.
